frontend/src/components/Stores.vue

```diff
@@ -0,0 +1,197 @@
+              <span v-if="s.readMoreActivated" v-html="s.description"></span>
+            </info-or-null>
+          </td>
+          <td class="col-3">
+            <info-or-null :condition="s.name">
+              <div class="row d-flex justify-content-center" style="max-width: 200px">
+                <input class="col-8 text-center" v-model="s.name">
+                <div class="col-4">
+                  <button class="btn btn-sm btn-outline-dark" @click="update(s)">Update</button>
+                </div>
+              </div>
+            </info-or-null>
+          </td>
+          <td class="col-2">
+            <info-or-null :condition="s.openingDate"><p style="text-align: center">{{ s.openingDate }}</p>
+            </info-or-null>
+          </td>
+          <td class="col-1" style="text-align: center">
+            <info-or-null :condition="s.storeType">{{ s.storeType }}</info-or-null>
+          </td>
+          <td class="col-1">
+            <info-or-null :condition="hasSeasons(s)">
+              <ul>
+                <li v-for="season in s.seasons" :key="season.id">
+                  {{ season.season }}
+                </li>
+              </ul>
+            </info-or-null>
+          </td>
+          <td class="col-2">
+            <info-or-null :condition="hasAdditionalInformation(s)">
+              <ul>
+                <li v-if="isValidField(s.additionalInfo.specialField1)">
+                  <p>{{ s.additionalInfo.specialField1 }}</p>
+                </li>
+                <li v-if="isValidField(s.additionalInfo.specialField2)">
+                  <p>{{ s.additionalInfo.specialField2 }}</p>
+                </li>
+              </ul>
+            </info-or-null>
+          </td>
+        </tr>
+        </tbody>
+      </table>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import VPagination from "@hennge/vue3-pagination";
+import "@hennge/vue3-pagination/dist/vue3-pagination.css";
+import InfoOrNull from "./InfoOrNull"
+
+export default {
+  name: 'Stores',
+  components: {
+    VPagination,
+    InfoOrNull
+  },
+  methods: {
+    resetFilter: function () {
+      this.nameFilter = ""
+      this.fetchStoresPage(this.page)
+    },
+    hasSeasons: function (store) {
+      return Array.isArray(store.seasons) && store.seasons.length
+    },
+    isValidField: function (f) {
+      return f && f.length > 0
+    },
+    hasAdditionalInformation: function (store) {
+      return store.additionalInfo && (this.isValidField(store.additionalInfo.specialField1) || this.isValidField(store.additionalInfo.specialField2))
+    },
+    activateReadMore(store) {
+      store.readMoreActivated = !store.readMoreActivated;
+    },
+    updatePageHandler: function (page) {
+      this.fetchStoresPage(page)
+    },
+    fetchStoresPage: function (page) {
+      this.stores = []
+      fetch(`/api/getStores?page=${page - 1}&storeNameFilter=${encodeURIComponent(this.nameFilter)}`)
```

Comment:
what's the rationale behind having API requests being declared directly on the component?

Reply:
None :(. Should have moved it to a separate service.